### ISSUE TITLE
[beta] Unable to get any challenges from a superblock that I did not start with in the first place.

### ISSUE THREAD

**Issue by raisedadead** [freeCodeCamp/freeCodeCamp#16803]
Unable to get any challenges from a superblock that I did not start with in the first place.

1. Started with ES6, and then signed in on fresh instance of the browser.
2. Went to the HTML super block.
3. Clicked through a few challenges.
4. I am getting the empty state.

![image](https://user-images.githubusercontent.com/1884376/36829210-1a35a2a4-1d44-11e8-9265-0b83c284cae2.png)

**Comment by raisedadead** [comment]
Probably relates to a recent splitting of the superblocks?

**Comment by Bouncey** [comment]
@raisedadead are you getting anything weird in the terminal? Something like `getting challenges for undefined block`? 

Do you see debug output for both the mapUi and challenge service for the page without challenge descriptions?

**Comment by Bouncey** [comment]
So you are on the live beta site?

Re-do your steps to reproduce with the network tab open, you should see a request to

`/services/challenge;blockName=${theBlockYouJustClicked};lang=en...`

Has that resolved by the time you click on a challenge?

**Comment by raisedadead** [comment]
Here is how its playing out on beta:

1. Visit a previous challenge from my solved challenges for example:
https://beta.freecodecamp.org/en/challenges/es6/compare-scopes-of-the-var-and-let-keywords

   Network Log:
   ![image](https://user-images.githubusercontent.com/1884376/36842306-4e237d2a-1d71-11e8-9c25-99d429c0db52.png)

   Which is:
   `https://beta.freecodecamp.org/services/challenge;blockName=es6;lang=en?_csrf=[****redacted****]&returnMeta=true`

2. Visit the HTML5 superblock, click any of the challenges

    Network Log:
    ![image](https://user-images.githubusercontent.com/1884376/36842489-d1ea373e-1d71-11e8-92e3-49272db22464.png)

    Each `update-my-current-challenge` is for a click that I made on different challenges on that section.


This means that the `/services/challenge;blockName...` request is never made again.

**Comment by Bouncey** [comment]
A quick update. I have reproduced this locally by throttling the network and selecting a challenge in a different block to my initial block. 

If I click the challenge before the challenge service has resolved, I get the `Happy coding` placeholder text.

What is strange is that, after the challenge service has resolved, the challenge description/seed does not update. 

I have created a loader to inform the user that the UI isn't quite ready for them. I just need to get the new challenge data to render when ready
